analyses/malawi/dryspells_trigger/07_ecmwf_monthly_skill_general.md

```diff
@@ -41,164 +37,372 @@ reload(processing)
-              ylabel="CRPS [mm]"):
-    fig, ax = plt.subplots()
-    df = df_crps.copy()
-    for i, subset in enumerate([k for k in df_crps.keys() if "crps" in k]):
-        y = df[subset]
-        if division_key is not None:
-            dkey = f'{division_key}_{subset.split("_")[-1]}' if subset!="crps" else division_key
-            y /= df[dkey]
-        ax.plot(df['leadtime'], y, ls="-", c=f'C{i}')
-    ax.plot([], [], ls="-", c='k')
-    # Add colours to legend
-    for i, subset in enumerate([k for k in append_dict.keys() if "crps" in k]):
-        ax.plot([], [], c=f'C{i}', label=subset)
-    ax.set_title("ECMWF forecast skill in Malawi:\n 2000-2020 forecast")
-    ax.set_xlabel("Lead time (months)")
-    ax.set_ylabel(ylabel)
-    ax.legend()
-```
-
-```python
-#performance pretty bad.. especially looking at the mean values, it is about 20% off on average for decjanfeb..
-# Plot absolute skill
-plot_skill(df_crps)
-
-# Rainy season performs the worst, but this is likely because 
-# the values during this time period are higher. Try using 
-# reduced skill (dividing by standard devation).
-plot_skill(df_crps, division_key='std', ylabel="RCRPS")
-
-#This is perhpas not exactly what we want because we know this 
-#data comes from the same location and the dataset has the same properties, 
-#but we are splitting it up by mean value. Therefore try normalizing using mean
-plot_skill(df_crps, division_key='mean', ylabel="NCRPS (CRPS / mean)")
+Below we plot the range of observed and forecasted values per month, but across all leadtimes. We can see that they both show the same pattern across the months. However, the observed data shows a broader range of values, and especially has more outliers. The medians are relatively in the same range for observed and forecasted values, though when zooming in you can see that they also differ. 
+
+```python
+fig, axes = plt.subplots(1,2,figsize=(20,8),sharey=True)
+sns.boxplot(data=df_forobs,x="month",y="precip_obs",ax=axes[0],color=hdx_blue)
+sns.boxplot(data=df_forobs,x="month",y="precip_for",ax=axes[1],color=hdx_blue)
+axes[0].set_title("Observed")
+axes[1].set_title("Forecasted")
+
+for ax in axes:
+    ax.set_ylabel("Monthly precipitation")
+    ax.set_xlabel("Month number")
+    ax.spines['right'].set_visible(False)
+    ax.spines['top'].set_visible(False)
+    ax.tick_params(axis='y', labelleft=True)
+fig.suptitle("Ranges of precipitation for different months");
+```
+
+Next we look at the bias. We firstly plot the observed vs forecasted-observed values across all leadtimes, dates, and cells. 
+From this we can see that
+
+- Most months with very low precipitation were correctly classified. 
+- Months with less than 300mm have the tendency to be overpredicted, i.e. we see a positive bias
+- Months with more than 300mm have the tendency to be underpredicted, i.e. we see a negative bias
+
+```python
+#plot the observed vs forecast-observed to get a feeling for the discrepancy between the two
```

Comment:
It's also interesting to plot just forecast vs observed. It just looks like a giant blob, which is not good. For Jan Feb you can see that the forecast histogram is shifted to higher values than the observed one.

Reply:
![image](https://user-images.githubusercontent.com/12556634/127837494-823fc317-7548-433d-8259-55c51ecb97c0.png)
 I agree it is a giant blob, but don't think it gets better when plotting for vs obs? (see image). Would it be an idea to remove all observed values below 20 mm  as that is the majority of points while we are not really interested in them? Or is that cheating

Reply:
Is what's plotted there really forecast vs obs? Anyway I was just curious to make that plot to see if there is any obvious relation between the two, or if it's more just random looking, which means the forecast isn't forecasting much of anything. 

> Would it be an idea to remove all observed values below 20 mm as that is the majority of points while we are not really interested in them?

I thought we are interested in these?